xmtp_api_grpc/src/grpc_api_helper.rs

```diff
@@ -32,8 +32,11 @@ use xmtp_proto::{
 async fn create_tls_channel(address: String) -> Result<Channel, Error> {
     let channel = Channel::from_shared(address)
         .map_err(|e| Error::new(ErrorKind::SetupError).with(e))?
+        .initial_connection_window_size(Some((1 << 31) - 1))
         .keep_alive_while_idle(true)
-        .connect_timeout(Duration::from_secs(5))
+        .connect_timeout(Duration::from_secs(10))
+        .tcp_keepalive(Some(Duration::from_secs(15)))
+        .timeout(Duration::from_secs(120))
```

Comment:
**Purpose:** Sets a maximum duration for the client to wait for a response to a request.

**Functionality:** If a response is not received within the specified timeout, the request is canceled and an error is returned.

**Impact:** This is critical for bounding the wait time for operations, which can enhance the predictability and reliability of client interactions by avoiding indefinitely hanging requests.